### PR TITLE
Viewer Multi Select

### PR DESCRIPTION
**Pull request by jameskerr**
Relates to #926 

The viewer is now able to select rows via "click", "cmd+click", and "shift+click".

image

Since multi-select doesn't make sense in the normal log search viewer at the moment (it would just create more work that is not priority), we will disable it there. It will be enabled in the archive search results table.

Things to finish before merging:

- [x] Adding a handler for "select all"
- [x] Changing the background color of the selected cell
- [x] Disabling it on the main viewer
- [x] Writing migration to remove "viewer" slice of state from being persisted
- [x] Removing "viewer" slice in the root reducer (it's just hanging out there doing nothing)

### CODE REVIEW

File: src/js/state/Tabs/reducer.js [hunk 1]
```diff
@@ -119,3 +120,14 @@ function updateTab(state, action: Object) {
 function indexOf(data, id) {
   return data.findIndex((t) => t.id === id)
 }
+
+function reduxAction(action) {
+  return action.type.startsWith("@@")
```
Comment (mason-fish) [inline src/js/state/Tabs/reducer.js:125]: I wasn't aware that this is associated with an internal redux event. Would you mind adding a little comment here providing some color on what this is doing, for posterity?

File: src/js/state/Viewer/reducer.js [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+  scrollPos: {x: 0, y: 0},
+  selection: {
+    rows: {},
+    currentRange: [0, 0]
+  }
+})
 
 export default function(
-  state: ViewerState = init,
+  state: ViewerState = init(),
```
Comment (mason-fish) [inline src/js/state/Viewer/reducer.js:21]: 👍 In addition to being careful about ensuring initial state does not reuse a pointer to an old object, the changes here are reminding me that we should prob choose to continue to adopt `immer` with the same policy we've decided for `styled-components`

File: src/js/state/Viewer/actions.js [hunk 3]
```diff
@@ -8,42 +8,88 @@ import type {
...
+
+export const selectMulti = (index: number): VIEWER_SELECT_MULTI => {
+  return {type: "VIEWER_SELECT_MULTI", index}
+}
+
+export const selectRange = (index: number): VIEWER_SELECT_RANGE => {
+  return {type: "VIEWER_SELECT_RANGE", index}
+}
+
+export const selectRangeNext = () => ({
```
Comment (mason-fish) [inline src/js/state/Viewer/actions.js:77]: Why no return type annotations for the remaining actions here?

File: src/js/state/Viewer/reducer.js (hunk 2 above)
Comment (jameskerr) [reply src/js/state/Viewer/reducer.js:21]: Yes, that makes sense to me.

File: src/js/state/Viewer/actions.js (hunk 3 above)
Comment (jameskerr) [reply src/js/state/Viewer/actions.js:77]: Oops, forgot to add those.

**Pull request merged**